IdentificationManager.js

```diff
@@ -10,26 +10,41 @@ const request = async (url, data, method = "POST") => {
         headers["X-CSRFToken"] = data.csrfmiddlewaretoken;
         body = JSON.stringify(data || {});
     }
-    try {
-        const response = await fetch(url, { method, headers, body });
 
-        const responseText = await response.text();
-
-        try {
-            const data = JSON.parse(responseText);
-            data.success = data.status === "success";
-            data.pending = `${response.status}` === "202";
-            return {
-                data,
-                ok: response.ok,
-            };
-        } catch (err) {
-            console.log("Failed to parse response as JSON", responseText);
-            return {};
+    const onError = async (err) => {
+        const retriesRemaining = retries - 1;
+        if (retriesRemaining > 0) {
+            console.log(`Error fetching ${url}: ${err}, waiting for 1000ms before retrying.`);
+            await new Promise((resolve) => setTimeout(resolve, 1000));
+            console.log(`Retrying ${url}, ${retriesRemaining} tries remaining.`);
+            return await request(url, data, method, retriesRemaining);
         }
-    } catch (err) {
         console.log(err);
         return {};
+    };
+
+    try {
+        const response = await fetch(url, { method, headers, body });
+        if (`${response.status}` !== "410") {
```

Comment:
nitpick but maybe doing it the other way around and using an early return for 410 would be easier to read.